### PR TITLE
clusters: add Labels widget to the MAPI cluster detail page

### PR DESCRIPTION
**Pull request by axbarsan**
This PR adds support for listing/editing/adding/removing cluster labels via the Management API in the new UI.

The implementation is fairly straight-forward, since it is based on the existing label editor UI. This PR also slightly improves the keyboard interaction with label management, and it makes the label deletion button's click target slightly bigger, so it's harder to miss it.

## Preview

<details>
<summary>No labels</summary>

image

</details>

<details>
<summary>Some labels</summary>

image

</details>

### CODE REVIEW

File: src/components/Cluster/ClusterDetail/ClusterLabels/ClusterLabels.tsx [hunk 1]
```diff
@@ -60,38 +50,45 @@ const NoLabels = styled.div`
...
+  border-bottom: 1px dotted;
 `;
 
+interface IClusterLabelsProps
+  extends Omit<ComponentPropsWithoutRef<'div'>, 'onChange'> {
+  onChange: (patch: ILabelChange) => void;
+  labels?: Record<string, string>;
+  isLoading?: boolean;
+  errorMessage?: string;
+  showTitle?: boolean;
```
Comment (axbarsan) [inline src/components/Cluster/ClusterDetail/ClusterLabels/ClusterLabels.tsx:66]: Making the component controllable, so I can re-use it for the MAPI implementation

File: src/components/Cluster/ClusterDetail/ClusterLabels/DeleteLabelButton.tsx [hunk 2]
```diff
@@ -35,12 +41,28 @@ const DeleteLabelButton: FC<IDeleteLabelButton> = ({
...
+     * Focus the cancel button after the confirmation is opened and
+     * all the microtasks have finished running.
+     * */
+    setTimeout(() => {
+      if (!isOpen || !cancelButtonRef.current) return;
+
+      const cancelButtonElement = cancelButtonRef.current.querySelector<HTMLButtonElement>(
+        '.cancel-button'
+      );
+      cancelButtonElement?.focus();
```
Comment (axbarsan) [inline src/components/Cluster/ClusterDetail/ClusterLabels/DeleteLabelButton.tsx:62]: Make the cancel button selected by default when the deletion tooltip is opened

File: src/components/MAPI/clusters/ClusterDetail/ClusterDetailOverview.tsx [hunk 3]
```diff
@@ -97,6 +98,60 @@ const ClusterDetailOverview: React.FC<{}> = () => {
     [orgId, clusterId]
   );
 
+  const labels = useMemo(() => getVisibleLabels(cluster), [cluster]);
+
+  const [labelsError, setLabelsError] = useState('');
+  const [labelsIsLoading, setLabelsIsLoading] = useState(false);
+
+  const handleLabelsChange = async (patch: ILabelChange) => {
```
Comment (axbarsan) [inline src/components/MAPI/clusters/ClusterDetail/ClusterDetailOverview.tsx:106]: Once this file gets larger I will probably extract all this logic into custom React hooks, but not worrying about that right now

File: src/components/UI/Display/MAPI/clusters/ClusterDetail/ClusterDetailWidgetOptionalValue.tsx [hunk 4]
```diff
@@ -40,17 +43,23 @@ const ClusterDetailWidgetOptionalValue: React.FC<IClusterDetailWidgetOptionalVal
     return <NotAvailable />;
   }
 
+  const refreshableKey = JSON.stringify(value);
```
Comment (axbarsan) [inline src/components/UI/Display/MAPI/clusters/ClusterDetail/ClusterDetailWidgetOptionalValue.tsx:46]: Silly way of serializing data structures, so `RefreshableLabel` would know if a value changed

**Review by marians — APPROVED** [review]
Yep, works! 🌻

**Pull request merged**